File: network/src/main/kotlin/org/bitlap/network/core/SessionFactory.kt
package org.bitlap.network.core

/**
 * Desc: create session impl from spi
 *
 * Mail: chk19940609@gmail.com
 * Created by IceMimosa
 * Date: 2021/9/4
 */
interface SessionFactory {

    fun create(username: String, password: String, sessionConf: Map<String, String>, sessionManager: SessionManager): Session
}

File: network/src/main/scala/org/bitlap/network/session/SessionFactory.scala
/* Copyright (c) 2022 bitlap.org */
package org.bitlap.network.session

/**
 * @author 梦境迷离
 * @since 2021/11/20
 * @version 1.0
 */
trait SessionFactory {

  def create(
    username: String,
    password: String,
    sessionConf: Map[String, String],
    sessionManager: SessionManager
  ): Session

}
